api/tokens/serializers.py

```diff
@@ -1,17 +1,46 @@
 from rest_framework import serializers as ser
 from rest_framework import exceptions
 
-from framework.auth.oauth_scopes import public_scopes
 from osf.exceptions import ValidationError
-from osf.models import ApiOAuth2PersonalToken
+from osf.models import ApiOAuth2PersonalToken, ApiOAuth2Scope
 
 from api.base.exceptions import format_validation_error
-from api.base.serializers import JSONAPISerializer, LinksField, IDField, TypeField
+from api.base.serializers import JSONAPISerializer, LinksField, IDField, TypeField, RelationshipField, StrictVersion
+
+
+class TokenScopesRelationshipField(RelationshipField):
+
+    def to_internal_value(self, data):
+        return {'scopes': data}
 
 
 class ApiOAuth2PersonalTokenSerializer(JSONAPISerializer):
     """Serialize data about a registered personal access token"""
 
+    def __init__(self, *args, **kwargs):
+        super(ApiOAuth2PersonalTokenSerializer, self).__init__(*args, **kwargs)
+
+        request = kwargs['context']['request']
+
+        # Dynamically adding scopes field here, depending on the version
+        if expect_scopes_as_relationships(request):
+            field = TokenScopesRelationshipField(
+                related_view='tokens:token-scopes-list',
+                related_view_kwargs={'_id': '<_id>'},
+                always_embed=True,
+                read_only=False,
+            )
+            self.fields['scopes'] = field
+            self.fields['owner'] = RelationshipField(
```

Comment:
Unlike `scopes`, there is no model change for `owner` in `ApiOAuth2PersonalToken`.

* Please correct me if wrong: I think this is a favorable improvement 👍  **independent of** `scopes` being relationships now. Unlike `name` which is a char field, `owner` is a foreign key / relationship.